controllers/ControllerPost.php

```diff
@@ -15,44 +15,31 @@ class ControllerPost {
 	private string $_server;
 
 
-	public function __construct( $slug, $twig) {
-		$this->_slug = $slug[2];
+	public function __construct(object $twig, int $slug) {
+		$this->_slug = $slug;
 		$this->_twig = $twig;
 		$this->_server = ( new ConstantGlobal(ServerRequest::fromGlobals()) )->getServerName()['SERVER_NAME'];
-		$this->renderPost();
 
 	}
 
-	public function renderPost(){
+	public function viewPost(){
 
 		$post = new PostsManager();
 		$comments = new CommentsManager();
 		$thePost = $post->getPost($this->_slug);
 		$allComments = $comments->postComment($this->_slug);
-/*		foreach ($allComments as  $value){
-			if (is_array($value)){
-				foreach ($value as $key => $values){
-					if ($values === 'content'){
-						$values = html_entity_decode($values);
-					}
-
-					echo $key." : ".$values.'<br>';
-				}
-			}
-		}
-		var_dump($allComments);*/
 		$keyPost = 'post';
 
 		(new Session)->setKey($keyPost, ['id' => $this->_slug]);
 		$key = (new Session)->getKey('user');
-
 		if (!empty($key)){
 			$modifComment = $key['id'];
 			echo $this->_twig->render( 'post.twig', ['thePost' => $thePost,
-			                                        'allComments' => $allComments,
-			                                        'logged'=> TRUE,
-			                                        'user'=> $key['pseudo'],
+			                                         'allComments' => $allComments,
+			                                         'logged'=> TRUE,
+			                                         'user'=> $key['pseudo'],
 			                                         'server' => $this->_server,
+			                                         'admin' => $key['admin'],
```

Comment:
Codacy found an issue: [All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$key['admin']'.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=6061935)